Require webpack_config field to build with custom configuration

EverlastingBugstopper opened the pull request:
Fixes #296 and #833 

As promised - we are deprecating inferred usage of `webpack.config.js` in 1.6.0 and are requiring the `webpack_config` file to be set in order to use a custom webpack configuration file. 

```console
$ wrangler build
⚠️  If you would like to use your own custom webpack configuration, you will need to add this to your wrangler.toml:
webpack_config = "webpack.config.js"
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build
using custom
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build --env test
⚠️  If you would like to use your own custom webpack configuration, you will need to add this to your wrangler.toml:
webpack_config = "webpack.config.js"
✨  Built successfully, built project size is 535 bytes.
```

```console
$ wrangler build --env test
using custom config
✨  Built successfully, built project size is 535 bytes.
```
      
```console
$ wrangler build
⚠️  Workers Sites does not support custom webpack configuration files
✨  Built successfully, built project size is 11 KiB.
```

ashleymichal commented on tests/build.rs:
```diff
@@ -139,21 +143,8 @@ fn it_fails_with_multiple_webpack_configs() {
-}
-
-#[test]
-fn it_fails_with_multiple_specify_webpack_configs() {
```
should this test still exist? it's kind of redundant but i don't think its outcome is changed by this PR.

EverlastingBugstopper replied:
It's the same test as `it_fails_with_multiple_webpack_configs` the only difference previously was one had inferred `webpack.config.js` existed and one had it in the toml - after this breaking change they are all "specify" tests

EverlastingBugstopper commented:
> the utils removal is a step backwards, kind of, but makes the status quo consistent so is fine.

I brought back the util and moved a couple of other tests to it - the reason I removed it earlier was because I didn't realize that I could use the webpack config file created programatically in the wrangler.toml

motiejunas commented:
Could you give more details on "Workers Sites does not support custom webpack configuration files"? Basically it means that I cannot use TypeScript for Workers Sites anymore?

ashleymichal commented:
@motiejunas the error message is historical; when wrangler implicitly looked at the root for a custom webpack config, it made sense to disable the functionality for sites since the root would likely contain a config for the app you were deploying, not for the workers site. with this change, though, we should allow custom webpack configs since you are required to declare them in your wrangler.toml. we'll track it in a separate issue.
